napari/layers/base/base.py

```diff
@@ -24,6 +27,48 @@
 Extent = namedtuple('Extent', 'data world step')
 
 
+def _find_layer_source() -> Source:
```

Comment:
Very nice!! I like it, and having a centralized place where this is all defined. Before merge though can you move this into a the layer utils folder, maybe a file named `napari/layers/utils/source_utils.py`.

Reply:
One concern I have now reading this is that `inspect.stack()` takes >80ms on my machine even on the base IPython shell (ie not a huge stack). We are already having issues with many layers, and now this code will run with every layer added. So adding 100 layers (we already have quite a few users doing this on a regular basis) costs 8s. So, I think we should balance the benefit of centralization of this method with the performance hit.